File: in_Java/0069 Sqrt(x).java
class Solution {
    public int mySqrt(int x) {
        if(x == 0) return 0;
        int i = 1;
        int j = x;
        int m;
        while(i <= j){
            m = i + (j - i) / 2;
            if(x / m >= m && (m+1) > x / (m+1)) return m;
            else if(m < x / m) i = m+1;
            else j = m-1;
        }
        return -1;
    }
}

File: in_Python/0069 Sqrt(x).py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        i = 1
        j = x/2
        if x == 1:
            return 1
        while i+1 < j:
            m = (j - i)/2 + i
            if m*m < x:
                i = m
            else:
                j = m
        if j * j <= x:
            return j
        else:
            return i
        